cmake/ccache.cmake

```diff
@@ -17,8 +19,11 @@ if(NOT MSVC)
       )
       set(WITH_CCACHE ON)
     elseif(WITH_CCACHE)
-      list(APPEND CMAKE_C_COMPILER_LAUNCHER ${CCACHE_EXECUTABLE})
-      list(APPEND CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE_EXECUTABLE})
+      foreach(lang IN ITEMS C CXX OBJCXX)
+        set(CMAKE_${lang}_COMPILER_LAUNCHER
```

Comment:
Why is this now overriding the variable instead of appending?

Reply:
Thanks!

I do not really have an explanation for this change...

We definitely should honour user-provided `CMAKE_<LANG>_COMPILER_LAUNCHER` environment variables if any.

The appending has been restored.

Reply:
Reverted [back](https://github.com/bitcoin/bitcoin/pull/30861#issuecomment-2645380742) given this [comment](https://github.com/bitcoin/bitcoin/pull/30861#discussion_r1941284740).